cloudinit/util.py

```diff
@@ -3231,8 +3229,8 @@ def deprecate(
 
     Note: uses keyword-only arguments to improve legibility
     """
-    if not hasattr(deprecate, "_log"):
-        deprecate._log = set()  # type: ignore
+    if not deprecate.__dict__.get("log"):
```

Comment:
[modifying __dict__ directly is not recommended](https://docs.python.org/3/library/stdtypes.html#modules). Why not `getattr` and `setattr` here, further down in this file, and in conftest.py?